botocore/handlers.py

```diff
@@ -99,12 +101,6 @@
 S3_SIGNING_NAMES = ('s3', 's3-outposts', 's3-object-lambda')
 VERSION_ID_SUFFIX = re.compile(r'\?versionId=[^\s]+$')
 
-SERVICE_NAME_ALIASES = {'runtime.sagemaker': 'sagemaker-runtime'}
-
-
-def handle_service_name_alias(service_name, **kwargs):
-    return SERVICE_NAME_ALIASES.get(service_name, service_name)
-
```

Comment:
For reusing the map, I would recommend:

1. Keep with the change of moving `SERVICE_NAME_ALIASES` to `utils`
2. In `utils` document it similar to the `CLIENT_NAME_TO_HYPHENIZED_SERVICE_ID_OVERRIDES` map where it is not part of botocore's public interface and also document what exactly it is supposed to represent for clarity.
3. Do not move `handle_service_name_alias` to the `utils` module and keep it defined here.
4. In the config provider, just use the `SERVICE_NAME_ALIASES` map from `utils`.

The main reason I'm suggesting this is that it does not make a lot of sense to move the `handle_service_name_alias` to `utils` as we are not really generalizing it to be a public utility and I don't think we should be making it a public utility as there is not a strong reason to do so. Referencing from the alias map is straightforward enough and the `handle_service_name_alias` only exists so we encapsulate the logic in the format of a botocore event handler.